include/boost/histogram/utility/agresti_coull_interval.hpp

```diff
@@ -0,0 +1,47 @@
+// Copyright 2022 Jay Gohil, Hans Dembinski
+//
+// Distributed under the Boost Software License, version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_HISTOGRAM_UTILITY_AGRESTI_COULL_INTERVAL_HPP
+#define BOOST_HISTOGRAM_UTILITY_AGRESTI_COULL_INTERVAL_HPP
+
+#include <boost/histogram/utility/binomial_proportion_interval.hpp>
+#include <cmath>
+#include <utility>
+
+namespace boost {
+namespace histogram {
+namespace utility {
+
+template <class ValueType>
+class agresti_coull_interval : public binomial_proportion_interval<ValueType> {
+  using base_t = binomial_proportion_interval<ValueType>;
+
+public:
+  using value_type = typename base_t::value_type;
+  using interval_type = typename base_t::interval_type;
+
+  explicit agresti_coull_interval(deviation d = deviation{1.0}) noexcept
+      : z_{static_cast<double>(d)} {}
+
+  interval_type operator()(value_type successes, value_type failures) const noexcept {
+    const double half{0.5}, one{1.0}, zsq = z_ * z_;
```

Comment:
if you use declare them to be `double`, you don't need the constants `half` and `one`, you can just use the numbers in the formula directly. But after switching back and forth, I think it is correct to replace `double` with `value_type` here.

What we really want is to ensure that `value_type` is a floating_point number. The `value_type` of the interval is not necessarily the same value_type that the class `fraction` uses. The latter can be an integer, but the value_type of the interval classes must be of floating point type.

Reply:
This is already ensured by the base class `binomial_proportion_interval`.

Reply:
In other words, please revert this change.